### ISSUE TITLE
what is the official lua file for the suite osm2pgsql / renderd(cartocss) ?

### ISSUE THREAD

**Issue by ghost** [openstreetmap/osm2pgsql#843]
On openstreetmap-carto there is openstreetmap-carto.lua and in the osm2pgsl distibution style.lua.
They do different things , particularly with z_order. 

Style.lua is near to tagtransform-c but openstreetmap-carto.lua is fresher.

If it is style.lua there is one or two possible small optimizations : a break ( or a direct return ) in the double loop of filter_tags_generic() and an order inversion of the 2 blocks of filter_basic_tags_rel(). Not very important, the time killer of the lua process being the strings stack push/pop.

**Comment by pnorman** [comment]
osm2pgsql is designed to be used with different style files and different tag transforms. There is no "official" one. In this repo, style.lua is an example so stylesheet authors can write their own. openstreetmap-carto.lua and openstreetmap-carto.style are what the OpenStreetMap Carto authors have written for their stylesheet.

**Comment by sav123** [comment]
you close very quickly and don't address the question as usual

**Comment by pnorman** [comment]
> you close very quickly and don't address the question as usual

Could you rephrase the question?

**Comment by StyXman** [comment]
> one or two possible small optimizations : a `break` ( or a direct `return` ) in the double loop of `filter_tags_generic()` and an order inversion of the 2 blocks of `filter_basic_tags_rel()`.

**Comment by lonvia** [comment]
The question in the topic has been adequately answered. If there are separate issues, the poster should open a separate ticket. Closing again.
